geoh5py/data/visual_parameters.py

```diff
@@ -0,0 +1,146 @@
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import xml.etree.ElementTree as ET
+from warnings import warn
+
+from .text_data import TextData
+
+PARAMETERS = ["Colour"]
+ATTRIBUTES = ["tag", "text", "attrib"]
+
+
+class VisualParameters(TextData):
+    _xml: ET.Element | None = None
+
+    @property
+    def xml(self) -> ET.Element | None:
+        """
+        :obj:`str` XML string.
+        """
+        if self._xml is None and isinstance(self.values, str):
+            self.xml = self.values  # type: ignore
+
+        return self._xml
+
+    @xml.setter
+    def xml(self, values: str):
+        if not isinstance(values, str):
+            raise TypeError(f"Input 'values' for {self} must be of type {str}.")
+
+        try:
+            self._xml = ET.fromstring(values)
+        except ET.ParseError as exception:
+            raise ValueError(
+                f"Input 'values' for {self} must be a valid XML string ({exception})."
+            ) from exception
+
+    @property
+    def colour(self) -> None | str | dict:
+        """
+        :obj:`dict` the Colour dictionary.
+        """
+        return self.get_child("Colour", "text")
+
+    @colour.setter
+    def colour(self, values: str):
+        # todo: check if values has the right format: what is this format???
+        if not isinstance(values, str):
+            raise TypeError(f"Input 'values' for {self} must be of type {str}.")
+
+        self.modify_xml({"tag": "Colour", "attrib": {}, "text": values})
+
+    def modify_xml(self, values: dict):
+        """
+        Modify the XML Element with a dictionary.
+        :param values: a dictionary of the values to modify
+        that must contains 'tag', 'attrib' and 'text' keys.
+        """
+        if not isinstance(values, dict):
+            raise TypeError(f"Input 'values' for {self} must be of type {dict}.")
+
+        self.set_child(values["tag"], "text", values["text"])
+        self.set_child(values["tag"], "attrib", values["attrib"])
+
+    def check_child(self, child: str, attribute: str) -> bool:
+        """
+        Check if the child is a valid parameter.
+        :param child: the key of the child to check.
+        :param attribute: the attribute to check.
+        :return: True if the child and attribute are valid.
+        """
+        if child not in PARAMETERS:
+            warn(f"{child} is not a valid parameter.")
+            return False
+
+        if not isinstance(self.xml, ET.Element):
+            warn(f"XML is not set for {self}.")
+            return False
+
+        if self.xml.find(child) is None:
+            warn(f"{child} is not present in the XML.")
```

Comment:
Maybe we should allow adding new ones if they are supported? For instance what would happen if Visual Parameter is not set initially?

Reply:
I don't think we should allow users to pass anything. 
Visual paramter is an already created object. 

We could have a function create_visual_parameters(**kwargs) somewhere to set up visual parameters if they are not created? 

However I agree we should allow the user to define an autorized parameter if it's not in the XML, I'll work on that.